externals/llvm-external-projects/torch-mlir-dialects/lib/Dialect/TMTensor/IR/TMTensorOps.cpp

```diff
@@ -403,8 +403,9 @@ bool ScatterOp::payloadUsesValueFromOperand(OpOperand *opOperand) {
     bbArgNumber = 0; // block arg 0 is `update`.
   else {
     bool isValidOperand = operand == indices() || operand == original();
-    assert(isValidOperand &&
-           "operand must belong to the current tm_tensor.scatter op");
+    if (!isValidOperand)
```

Comment:
the standard way to deal with these unused-if-no-asserts variables in LLVM is by casting to void such as:

```
bool foo = ...;
(void)foo;
assert(foo && ...)
```

Reply:
Thanks @silvasean!  Fixed in the most recent commit.